test/access/roles/WhitelistAdminRole.test.js

```diff
@@ -2,21 +2,82 @@ const { shouldBehaveLikePublicRole } = require('../../behaviors/access/roles/Pub
     await this.contract.addWhitelistAdmin(otherWhitelistAdmin, { from: whitelistAdmin });
   });
 
   shouldBehaveLikePublicRole(whitelistAdmin, otherWhitelistAdmin, otherAccounts, 'whitelistAdmin');
-  it('removes role from all other accounts but the whitelister', async function () {
-    await this.contract.resetWhitelist({ from: whitelistAdmin });
-    await shouldFail.reverting(this.contract.onlyWhitelistAdminMock({ from: otherWhitelistAdmin }));
-    await this.contract.onlyWhitelistAdminMock({ from: whitelistAdmin });
+});
+
+contract('WhitelistAdminRole', function ([whitelistAdmin, otherWhitelistAdmin, ...otherAccounts]) {
+  it('removes role from all other accounts but the whitelister', async () => {
+    const contract = await WhitelistAdminRoleMock.new({ from: whitelistAdmin });
+    await contract.addWhitelistAdmin(otherWhitelistAdmin, { from: whitelistAdmin });
+    await contract.addWhitelistAdmin(otherAccounts[0], { from: whitelistAdmin });
+    await contract.resetWhitelist({ from: whitelistAdmin });
+    await shouldFail.reverting(contract.onlyWhitelistAdminMock({ from: otherWhitelistAdmin }));
+    await shouldFail.reverting(contract.onlyWhitelistAdminMock({ from: otherAccounts[0] }));
+    await contract.onlyWhitelistAdminMock({ from: whitelistAdmin });
   });
-  it('Transferring ownership should also add to whitelist', async function () {
+
+  it('Transferring ownership should also add to whitelist', async () => {
     const contract = await WhitelistAdminRoleMock.new({ from: whitelistAdmin });
     await contract.transferOwnership(otherWhitelistAdmin, { from: whitelistAdmin });
     await contract.onlyWhitelistAdminMock({ from: otherWhitelistAdmin });
   });
 });
+
+const add18Admins = contract => ADDRESSES.slice(0, 18).map(address => contract.addWhitelistAdmin(address));
+
+contract.only('WhitlistAdminRole', function ([deployer, secondAccount]) {
+  it('should not be possible to add more than 20 admins', async () => {
+    const contract = await WhitelistAdminRoleMock.new({ from: deployer });
+
+    await contract.addWhitelistAdmin(secondAccount);
+
+    const addAdressesToWhitelistAdminsPromises = add18Admins(contract);
+    await Promise.all(addAdressesToWhitelistAdminsPromises);
+
+    await shouldFail.reverting(contract.addWhitelistAdmin(ADDRESSES[18], { from: deployer }));
+    await shouldFail.reverting(contract.addWhitelistAdmin(ADDRESSES[19], { from: deployer }));
+  });
+
+  it('should be possible to add more admins after resetting', async () => {
+    const contract = await WhitelistAdminRoleMock.new({ from: deployer });
+    await contract.addWhitelistAdmin(secondAccount);
+
+    const addAdressesToWhitelistAdminsPromises = add18Admins(contract);
+    await Promise.all(addAdressesToWhitelistAdminsPromises);
+
+    await contract.renounceWhitelistAdmin({ from: secondAccount });
+    await contract.resetWhitelist({ from: deployer });
+    await contract.addWhitelistAdmin(ADDRESSES[19], { from: deployer });
```

Comment:
Make this test pass and then report back. You should be able to implement a working solution. Afterwards we will discuss together a better solution.